contracts/asserts.cairo

```diff
@@ -0,0 +1,32 @@
+use contracts::dummy_syscalls;
+
+const CORRECT_TX_VERSION: felt = 1;
+const QUERY_VERSION: felt = 2;
+
+fn assert_only_self() {
+    let self = dummy_syscalls::get_contract_address();
+    let caller_address = dummy_syscalls::get_caller_address();
+    assert(self == caller_address, 'argent: only self');
+}
+
+fn assert_non_reentrant(signer: felt) {
+    let caller_address = dummy_syscalls::get_caller_address();
+    assert(caller_address == 0, 'argent: argent: no reentrant call');
+}
+
+
+fn assert_initialized(signer: felt) {
+    // read signer from contract storage 
+    // let signer = signer::read();
+    assert(signer != 0, 'argent: account not initialized');
+}
+
+fn assert_correct_tx_version(tx_version: felt) {
+    assert(tx_version == CORRECT_TX_VERSION ^ tx_version == QUERY_VERSION, 'argent: invalid tx version');
+}
+
+fn assert_guardian_set(guardian: felt) {
```

Comment:
this could probably be renamed to smthng more generic like assert_not_zero

Reply:
assert non zero might as well be inlined as `assert(value != 0, "custom message")` cause you usually want a more specific error than just "was zero"

Reply:
should i remove assert guardian / assert initliazed and leave them to be used inline instead ?

Reply:
I think it's ok for now so we can get this merged, can always factor things later on if needed